### PR TITLE
Data models of Open Targets data metadata implemented using pydantic for type driven JSON deserialisation

### PR DESCRIPTION
**Pull request by kpto**
Title says it all. The test is just trying to deserialise the metadata of `targets` dataset, the models are highly cohesive so a successful deserialisation is already a pass, no extra assertion is needed.

### CODE REVIEW

File: open_targets/data/discover/model.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+from typing import Annotated, Literal, TypeAlias
+
+from pydantic import BaseModel, ConfigDict, Field, field_validator
+from pydantic.alias_generators import to_camel
+
+
+class ConfiguredBaseModel(BaseModel):
+    """Base model with extra configuration.
+
+    Base model configured to convert names from camel case to snake case and
```
Comment (slobentanzer) [inline open_targets/data/discover/model.py:14]: It seems to me that the behaviour is the other way around, from snake case to camel case, no?

Comment (slobentanzer) [reply open_targets/data/discover/model.py:14]: Ah I think I got it now, the target *in Python* is snake case

Comment (kpto) [reply open_targets/data/discover/model.py]: I improved the docstring a bit, I also renamed the directory from `discover` to `metadata`.

**Comment by slobentanzer** [comment]
Looks great, good job :)

**Pull request merged**